Branch-SDK/src/main/java/io/branch/referral/PrefHelper.java

```diff
@@ -419,7 +421,16 @@ public void setRandomizedDeviceToken(String randomized_device_token) {
      * @return A {@link String} that uniquely identifies this build.
      */
     public String getRandomizedDeviceToken() {
-        return getString(KEY_RANDOMIZED_DEVICE_TOKEN);
+        // If a newly (5.1.4+) set, valid, value exists, return it
+        String rdt = getString(KEY_RANDOMIZED_DEVICE_TOKEN);
+        if(!TextUtils.isEmpty(rdt) && !rdt.equals(NO_STRING_VALUE)){
+            return rdt;
+        }
+        // Otherwise this call checks if we have the old "bnc_device_fingerprint_id",
+        // or "bnc_no_value" if neither the new, or old value do not exist
+        else {
+            return getString(KEY_DEVICE_FINGERPRINT_ID);
```

Comment:
does this return null when empty? or an empty string?

Reply:
I believe getString would return "bnc_no_value"